packages/playground/website/src/lib/make-blueprint.tsx

```diff
@@ -17,11 +18,19 @@ export function makeBlueprint(options: MakeBlueprintOptions): Blueprint {
 			wp: options.wp as any,
 		},
 		steps: [
-			{
-				step: 'login',
-				username: 'admin',
-				password: 'password',
-			},
+			options.import && /^(http(s?)):\/\//i.test(options.import)
+				? {
+						step: 'importFile',
+						file: {
+							resource: 'url',
+							url: options.import,
+						} as UrlReference,
+				  }
+				: {
+						step: 'login',
+						username: 'admin',
+						password: 'password',
+				  },
```

Comment:
while it makes sense that an import may have a different admin user, when we make this change it won't be easy for someone to auto-login to their imported site unless they add custom PHP to do so. I guess if they add an import they will need to manually add the login step (if they want it) and we should communicate that?

Reply:
> an import may have a different admin user

Exactly, that was the thinking behind why the initial step is either a site import *or* an auto-login step, since we don't know that the admin user credentials will be the same as the default empty site.

> when we make this change it won't be easy for someone to auto-login to their imported site unless they add custom PHP to do so.

True, and crucially, there's currently no way to auto-login to an imported site from the Query API (which can only have `login=1`).

A simplest naive solution would be, if we could assume that the admin user in the imported site has the same credentials as the default empty site. Then we can run import site *and* auto-login steps initially on load. If the admin user is different, then auto-login can only happen through a custom Blueprint step and not through the Query API (currently).

Reply:
> no way to auto-login to an imported site from the Query API

I guess we do have this: passing blueprints as a URL fragment, `#{"steps":[{"step":"login","username":"admin","password":"password"}]}`

so maybe it's not as bad as I first thought. again, a bit verbose, but something someone can already do and we can make a builder UI to generate an appropriate URL.

> If we had the concept of an optional ("fail-able") step, we could import a site then attempt to log in as default admin user, continuing to remaining steps even if the login fails.

this sounds great! and a good use of the optional steps

Reply:
@eliot-akira when the `login` step fails, it fails silently. That's not ideal and I'd love to introduce the concept of optional steps, but we could actually always try to log the user in as `admin`/`password` until the optional steps work lands. Would you please update this PR? I'd love to get this in.